serialized_data_interface/__init__.py

```diff
@@ -17,7 +17,19 @@
 ]
 
 
-class NoCompatibleVersions(Exception):
+class SDIError(Exception):
+    status = BlockedStatus
```

Comment:
What about instead doing something like:

```suggestion
    status_class = BlockedStatus
    
    @property
    def status(self):
        return self.status_class(str(self))
```

Reply:
agreed, the other examples (OCIImageResourceError and CheckFailed) have `.status` as a fully instantiated status, not a recommended class.  Either property or actual status is fine with me